packages/wallet-standard/src/register.ts

```diff
@@ -1,14 +1,71 @@
-import type { WalletsWindow } from "@wallet-standard/base";
-import { GlowWallet } from "./wallet.js";
-
-declare const window: WalletsWindow;
-
-export function register(): void {
-  try {
-    (window.navigator.wallets ||= []).push(({ register }) =>
-      register(new GlowWallet())
-    );
-  } catch (error) {
-    console.error(error);
-  }
+// This is copied from @wallet-standard/wallet
+
+import type {
+    DEPRECATED_WalletsWindow,
+    Wallet,
+    WalletEventsWindow,
+    WindowRegisterWalletEvent as WindowRegisterWalletEventInterface,
+    WindowRegisterWalletEventCallback,
+} from '@wallet-standard/base';
+
+export function registerWallet(wallet: Wallet): void {
+    const callback: WindowRegisterWalletEventCallback = ({ register }) => register(wallet);
+    try {
+        (window as WalletEventsWindow).dispatchEvent(new WindowRegisterWalletEvent(callback));
+    } catch (error) {
+        console.error('wallet-standard:register-wallet event could not be dispatched\n', error);
+    }
+    try {
+        (window as WalletEventsWindow).addEventListener('wallet-standard:app-ready', ({ detail: api }) =>
+            callback(api)
+        );
+    } catch (error) {
+        console.error('wallet-standard:app-ready event listener could not be added\n', error);
+    }
+}
+
+class WindowRegisterWalletEvent extends Event implements WindowRegisterWalletEventInterface {
+    readonly #detail: WindowRegisterWalletEventCallback;
+
+    get detail() {
+        return this.#detail;
+    }
+
+    get type() {
+        return 'wallet-standard:register-wallet' as const;
+    }
+
+    constructor(callback: WindowRegisterWalletEventCallback) {
+        super('wallet-standard:register-wallet', {
+            bubbles: false,
+            cancelable: false,
+            composed: false,
+        });
+        this.#detail = callback;
+    }
+
+    /** @deprecated */
+    preventDefault(): never {
+        throw new Error('preventDefault cannot be called');
+    }
+
+    /** @deprecated */
+    stopImmediatePropagation(): never {
+        throw new Error('stopImmediatePropagation cannot be called');
+    }
+
+    /** @deprecated */
+    stopPropagation(): never {
+        throw new Error('stopPropagation cannot be called');
+    }
+}
+
+/** @deprecated */
+export function DEPRECATED_registerWallet(wallet: Wallet): void {
+    registerWallet(wallet);
+    try {
+        ((window as DEPRECATED_WalletsWindow).navigator.wallets ||= []).push(({ register }) => register(wallet));
+    } catch (error) {
+        console.error('window.navigator.wallets could not be pushed\n', error);
+    }
```

Comment:
We preserve the existing behavior as well as the new behavior to avoid breaking any apps that have already updated to the latest RC of Wallet Adapter.